src/slider/__tests__/slider.test.js

```diff
@@ -0,0 +1,186 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+import React from 'react';
+import {mount} from 'enzyme';
+import {Slider, StyledThumb} from '../index';
+
+/* global document */
+/* global window */
+
+const originalDocumentAddListener = document.addEventListener;
+
+describe('Stateless slider', function() {
+  let wrapper;
+  let allProps: {
+      range: Array<number>,
+      value: Array<number>,
+      step: ?number,
+      onChange: () => void,
+    },
+    mockFn;
+
+  beforeEach(function() {
+    //$FlowFixMe
+    document.addEventListener = jest.fn();
+    mockFn = jest.fn();
+    allProps = {
+      onChange: mockFn,
+      range: [0, 100],
+      value: [60],
+      step: 0,
+    };
+  });
+
+  afterEach(function() {
+    jest.restoreAllMocks();
+    document.addEventListener.mockClear();
+    wrapper && wrapper.unmount();
+    //$FlowFixMe
+    document.addEventListener = originalDocumentAddListener;
+  });
+
+  it('should render slider component', function() {
+    wrapper = mount(<Slider {...allProps} />);
+    expect(wrapper).toMatchSnapshot('Component has correctly rendered');
+  });
+
+  describe('events', function() {
+    let event, thumb, step, axisSizeInPixels;
+    function emulateThumbDown(thumbIndex = 0) {
+      wrapper = mount(<Slider {...allProps} />);
+      event = {movementX: 0};
+      thumb = wrapper.find(StyledThumb).at(thumbIndex);
+      thumb.simulate('mousedown', event);
+    }
+    beforeEach(function() {
+      step = 10;
+      axisSizeInPixels = 300;
+      jest
+        .spyOn(window, 'getComputedStyle')
+        .mockImplementation(() => ({width: axisSizeInPixels}));
+      allProps.step = step;
+    });
+
+    it('should start tracking thumb move when user clicks on it', function() {
+      emulateThumbDown();
```

Comment:
here we should also check that it subscribed for the `mousemove` and `mouseup` events